### PR TITLE
Update link api follow up

### PR DESCRIPTION
**Pull request by anandtiwary**

### CODE REVIEW

File: projects/common/src/navigation/navigation.service.ts [hunk 1]
```diff
@@ -106,31 +102,20 @@ export class NavigationService {
...
-  }
-
-  /**
-   * Navigate to an external URL.
-   * URL should be of the form 'https://url' or 'http://url'
-   */
-  public navigateExternal(
-    url: string,
-    navigationType: NavigationType = NavigationType.SameWindow
+  public navigateWithinApp(
```
Comment (anandtiwary) [inline projects/common/src/navigation/navigation.service.ts:105]: @aaron-steinfeld  Should we rip this off as well? We kind of need a method that would build the NavigationParams from these three inputs

**Comment by anandtiwary** [comment]
Will add more test in navigation service and link component after initial feedback

File: projects/components/src/navigation/navigation-list.component.ts [hunk 2]
```diff
@@ -83,7 +83,7 @@ export class NavigationListComponent {
   }
 
   public navigate(item: NavItemLinkConfig): void {
-    this.navigationService.navigateWithinApp(item.matchPaths[0], { relativeTo: this.activatedRoute });
+    this.navigationService.navigateWithinApp(item.matchPaths[0], this.activatedRoute);
```
Comment (aaron-steinfeld) [inline projects/components/src/navigation/navigation-list.component.ts:86]: We could probably use ht-link for the main items too right? That way they get the real href behavior

File: projects/common/src/navigation/navigation.service.ts [hunk 3]
```diff
@@ -106,31 +102,20 @@ export class NavigationService {
...
+    return this.navigate({
+      navType: NavigationParamsType.InApp,
+      path: path,
+      queryParams: this.buildQueryParam(preserveParameters ?? []),
+      relativeTo: relativeTo
+    });
   }
 
-  public buildParamsForNavigation(preserveParameters: string[] = []): QueryParamObject {
+  private buildQueryParam(preserveParameters: string[] = []): QueryParamObject {
```
Comment (aaron-steinfeld) [inline projects/common/src/navigation/navigation.service.ts:118]: nit: buildQueryParams (or buildQueryParamObject)

File: projects/common/src/navigation/navigation.service.ts (hunk 1 above)
Comment (aaron-steinfeld) [reply projects/common/src/navigation/navigation.service.ts:105]: by rip off do you mean remove it? If so, I would agree - send everything through navigate. `NavigationParams` already supports the relative path. If the preserved params is still used we can add support for it, but I don't think it is from a quick search.

**Review by aaron-steinfeld — APPROVED** [review]

File: projects/common/src/navigation/navigation.service.ts (hunk 1 above)
Comment (anandtiwary) [reply projects/common/src/navigation/navigation.service.ts:105]: Yeah. I wasn't sure about it either. I will put up a separate PR to remove this method and its usages.

File: projects/components/src/navigation/navigation-list.component.ts (hunk 2 above)
Comment (anandtiwary) [reply projects/components/src/navigation/navigation-list.component.ts:86]: Yep. Didn't want to club it with this PR. @arjunlalb is also going to help with this.

File: projects/common/src/navigation/navigation.service.ts (hunk 3 above)
Comment (anandtiwary) [reply projects/common/src/navigation/navigation.service.ts:118]: Yes for buildQueryParamObject but chose the shorter form. I can update it in the next pr.

**Pull request merged**